### DISCUSSION TITLE
Machine removal

### DISCUSSION THREAD

**Discussion by OBattler** [86Box/86Box#1429] [Ideas]
So, I'm trying to decide on my end, which machines to remove. I have already removed the Rise R418 (but I might bring it back if there's enough popular demand for it), and I'm considering removing those UMC chipset machines as the chipset is reverse-engineered and no datasheet is available.

Should I bring back the Rise R418?

And should those UMC chipset machines be removed?

And what other machines, if any, do you think should be removed?

**Comment by waltje** [comment]
Why was the R418 removed?  It seems to work fine, I have several configs with it on my end..?
The UMCs: yeah, those I would side-line until more info becomes available..

Just my $0.03 ..

    On Tuesday, May 4, 2021, 9:15:18 PM EDT, Miran Grča ***@***.***> wrote:  
 
 


So, I'm trying to decide on my end, which machines to remove. I have already removed the Rise R418 (but I might bring it back if there's enough popular demand for it), and I'm considering removing those UMC chipset machines as the chipset is reverse-engineered and no datasheet is available.

Should I bring back the Rise R418?

And should those UMC chipset machines be removed?

And what other machines, if any, do you think should be removed?

**Reply by OBattler** [reply]
Becuase we have other SiS 496/497 machines with Award BIOS now - the Zida 4DPS and the Soyo 4SA2.

My rationale is: the Rise R418 has an Award 4.50G BIOS, just like the Abit AB-PB4, while the Zida 4DPS has a unique Award 4.51PG BIOS with the Funworld logo. Also, the Zida 4DPS was explicitly requested by a user, while the Rise R418 is more or less a legacy of PCem-X, where it was presented alongside the SiS 496/497 from PCem, that I have no idea what machine that even is.

**Comment by ghost** [comment]
Hmm I would go with that way
Hedaka HED-919: Undocumented behavior at how it handles upper memory
One of the Samsung SPC's. Except if there are differences
OpenAT: Doesn't even work
One of the Olivetti M300's
Rise Computer R418: Part of our PCem legacy. Already better boards exist for it
Soyo 4SA2: If there's no reason to keep it
Shuttle HOT-539: The UMC 8890 is badly written. Shadowing is confusing. Better it goes away
Azza 5IVG: Part of the LGS Prime3C implementation. Other than that opinions fully yours
Epox P55-VA: Part of our PCem legacy. Our current ASUS 430VX is better
Gigabyte GA-686BX: What's the reason we implemented it. Just because PCem has it that doesn't mean we have to gain it too :b

**Reply by OBattler** [reply]
> Hedaka HED-919: Undocumented behavior at how it handles upper memory
Agreed.

> One of the Samsung SPC's. Except if there are differences
I need to look at them.

> OpenAT: Doesn't even work
Does it still not work?

> One of the Olivetti M300's
Yeah, that can go.

> Rise Computer R418: Part of our PCem legacy. Already better boards exist for it
The first part is wrong (PCem has a different Award SiS 496/497, that we essentially replaced with the R418), but the second part I agree with.

> Soyo 4SA2: If there's no reason to keep it
I believe it's our only 486 machine that can boot from CD.

> Shuttle HOT-539: The UMC 8890 is badly written. Shadowing is confusing. Better it goes away
I agree.

> Azza 5IVG: Part of the LGS Prime3C implementation. Other than that opinions fully yours
I was about to remove it, but then I noticed what Super I/O chip it uses, which stopped me.

> Epox P55-VA: Part of our PCem legacy. Our current ASUS 430VX is better
We had the Epox P55-VA long before PCem. The PCem legacy is the Shuttle HOT-557, but that's useful for testing ISA PnP, since it has an ISA PnP Super I/O chip.

> Gigabyte GA-686BX: What's the reason we implemented it. Just because PCem has it that doesn't mean we have to gain it too :b
I agree.

**Comment by OBattler** [comment]
Also, should the ALi M1429(G) machines with AMIBIOS and AMI WinBIOS be removed? They kind of sort of work (but the M1429 seems to be a Green chipset, so it has SMM stuff that's not implemented, so it's far complete), but the chipset is purely reverse-engineered. The Acer V1G I'm willing to keep because, being our only 486 Acer, it's rather unique.

**Reply by ghost** [reply]
> 
> 
> Also, should the ALi M1429(G) machines with AMIBIOS and AMI WinBIOS be removed? They kind of sort of work (but the M1429 seems to be a Green chipset, so it has SMM stuff that's not implemented, so it's far complete), but the chipset is purely reverse-engineered. The Acer V1G I'm willing to keep because, being our only 486 Acer, it's rather unique.

We have the AOpen SiS 471 which is WinBios 1993 and has it's chipset documented so yes

**Reply by ppgrainbow** [reply]
It would be best to keep the Olystar LiL-1429 and the Acer V1G. Get rid of the unidentified "AMI WinBIOS".

**Comment by nerd73** [comment]
I think there's a better case for removing the Zida 4DPS than there is for the Rise R418, at least in my opinion. We currently already have two Award 4.51PG SiS 496/497 machines (and the Zida is a 4.51PG as well) and the R418 is currently our only Award 4.50G machine. Special logo aside, 4.50g vs 4.51pg is a big difference: One supports Plug & Play and the other does not.

**Comment by waltje** [comment]
A purely revved chipset in itself isnt a problem. A lot of hardware emulationis done that way.  If it doesn't work, then, well, maybe it should be removed,or marked.
I personally am having no issues with the ALi machines, so will keep em.
--Fred

    On Wednesday, May 5, 2021, 2:06:36 PM EDT, Panagiotis ***@***.***> wrote:  
 
 



Also, should the ALi M1429(G) machines with AMIBIOS and AMI WinBIOS be removed? They kind of sort of work (but the M1429 seems to be a Green chipset, so it has SMM stuff that's not implemented, so it's far complete), but the chipset is purely reverse-engineered. The Acer V1G I'm willing to keep because, being our only 486 Acer, it's rather unique.


We have the AOpen SiS 471 which is WinBios 1993 and has it's chipset documented so yes

**Comment by waltje** [comment]
OpenAT was mine, akin to OpenXT, aimed to get an opensource BIOS.  I got that,eventually, but the machine is basically not useful, so should be deleted. Until wefind a *usable* opensource AT BIOS.
--Fred

    On Wednesday, May 5, 2021, 11:31:08 AM EDT, Miran Grča ***@***.***> wrote:  
 
 



Hedaka HED-919: Undocumented behavior at how it handles upper memory
Agreed.



One of the Samsung SPC's. Except if there are differences
I need to look at them.



OpenAT: Doesn't even work
Does it still not work?



One of the Olivetti M300's
Yeah, that can go.



Rise Computer R418: Part of our PCem legacy. Already better boards exist for it
The first part is wrong (PCem has a different Award SiS 496/497, that we essentially replaced with the R418), but the second part I agree with.



Soyo 4SA2: If there's no reason to keep it
I believe it's our only 486 machine that can boot from CD.



Shuttle HOT-539: The UMC 8890 is badly written. Shadowing is confusing. Better it goes away
I agree.



Azza 5IVG: Part of the LGS Prime3C implementation. Other than that opinions fully yours
I was about to remove it, but then I noticed what Super I/O chip it uses, which stopped me.



Epox P55-VA: Part of our PCem legacy. Our current ASUS 430VX is better
We had the Epox P55-VA long before PCem.



Gigabyte GA-686BX: What's the reason we implemented it. Just because PCem has it that doesn't mean we have to gain it too :b
I agree.

**Comment by ppgrainbow** [comment]
I have a list of emulated machines that I might want to be kept, renamed or removed. However, the list is not complete, but if you have found any more, let me know. :)

Machines that are facing potential removal:

OpenAT: Does not work at all.
Hedaka HED-919: Undocumented behavour on the way how it handles upper memory.
IBM PS/2 Model 70 (Type 3): Throws 216 error (base 64 KB on daughter card #1 failed), I'm wondering what could be causing this.
[ALi 1429] AMI WinBIOS 486: Although the board is a Kaimei brand, I can't seem to properly identify the exact name even though the board works as is.
Any of the machines that use the UMC 8890 chipset such as the Shuttle HOT-539 - datasheet is poorly written.

Machines that have been identified and will need to be renamed:

[ACC 2168] AMI 386DX clone -> [ACC 2168] JUKO AT046DX3
[ISA] Micronics 386 clone -> [Discrete Logic A82385 / 82C206] Micronics 386 I-CACHE
[NEAT] Phoenix 286 clone -> [NEAT] Kinetec 286/12
[SiS 401] AMI 486 clone -> [SiS 401] Chaintech 4xxSX/4xxSC
[SiS 471] AMI 486 clone -> [SiS 471] EPoX GXA486SG
[SiS 471] AMI WinBIOS clone -> [SiS 471] ABit 486 VESA AG4
[SiS 460] ABit AV4 -> [SiS 460] ABit AB-AV4
[SiS 471] Phoenix SiS-471 -> [SiS 471] SiS VL-BUS 471 Revision .A1 (Unidentified)

Notes:

1. The ABit AB-AV4 supports up to 32 MB of RAM using two 4M x 9 (16 MB) memory modules on banks 0 and 1.
2. The EPoX GXA486SG supports up to 128 MB of RAM.
3. SiS VL-BUS 471 Revision .A1 maxes out at 128 MB of RAM; however, the exact brand of the motherboard remains unidentified.


Machines that I would like to keep:

[C&T 386] ECS 386/32: The machine maxes out at 8 MB of RAM on the system board and uses the C&T 8230 386/AT Chipset
[ALi 1429] Olystar LiL1429: Also known as Seritech 486-VLA, this is the board that apparently has received popular demand for emulating 486-based machines.
[UMC 8881] Shuttle HOT-433: I personally don't have any problems with the functionality of the emulated machine.

If there is anything that I have missed, please let me know. :)

**Comment by EngiNerd89** [comment]
I'm probably late in this discussion, but rather than removing machines that have issues but are to some extent usable, wouldn't be better to move them inside the dev (or another ad-hoc) branch? In this way, the stable release would contain only fully working machines, while at the same time those who are fine with emulating incomplete machines could still use them. Also, the progress made so far in emulating incomplete machines would be preserved, allowing others who are more skilled (or have more spare time) to improve those machines without having to start from scratch.

**Reply by OBattler** [reply]
I'm running into issues with that idea right now - I'm having to identify keyboard controllers for all the machines, and the sheer number of them is increasing the workload massively.

> allowing others who are more skilled (or have more spare time) to improve those machines without having to start from scratch.

Any code is always present in the repository's history and can as such be brought back at any time.

> OpenAT: Does not work at all.

It works almost fine on my end now, somehow, just DOS hangs after the splash screen (I was testing using IBM DOS 6.1 Danish)m no idea why, but I haven't yet exhausted my tools to handle that.

> Hedaka HED-919: Undocumented behavour on the way how it handles upper memory.

I agree this one can go.

> IBM PS/2 Model 70 (Type 3): Throws 216 error (base 64 KB on daughter card #1 failed), I'm wondering what could be causing this.

Perhaps my rewrite of memory access handling messed that up.

> [ALi 1429] AMI WinBIOS 486: Although the board is a Kaimei brand, I can't seem to properly identify the exact name even though the board works as is.

This one is indeed going to be removed.

> Any of the machines that use the UMC 8890 chipset such as the Shuttle HOT-539 - datasheet is poorly written.

These are going to be removed as well.

> [C&T 386] ECS 386/32: The machine maxes out at 8 MB of RAM on the system board and uses the C&T 8230 386/AT Chipset

This is going to stay as it works without problems.

> [ALi 1429] Olystar LiL1429: Also known as Seritech 486-VLA, this is the board that apparently has received popular demand for emulating 486-based machines.

The only reason it's in such demand is because PCem has it, but we'll see about it.

> [UMC 8881] Shuttle HOT-433: I personally don't have any problems with the functionality of the emulated machine.

This might end up being the only UMB 8881 machine that remains. But it's not guaranteed.

> Why was the R418 removed?  It seems to work fine, I have several configs with it on my end..?
I'm going to re-add that as I have changed my mind about it.

> A purely revved chipset in itself isnt a problem. A lot of hardware emulationis done that way.  If it doesn't work, then, well, maybe it should be removed,or marked.
I personally am having no issues with the ALi machines, so will keep em.

The thing is, as Tiseno100 pointed out, the ALi M1429 is a Green chipset, so it has all the power management support (including SMI# and SMRAM) that has not been reverse engineered at all and can never be, so it's a partial implementation at best, with no room for improvement. Then again, the same goes for the UM8881, so perhaps all machines with it should go as well, and so should the Acer V1G with the ALi M1429 chipset.

On that note, is there any other 486 Acer machine?

Also, the SiS 5598, VT 8601, and all associated machines are currently set to be removed as well. Both have on-board graphics, while we made an exception for the STPC SBC's, their on-board graphics is at least simple enough that it's feasiable to eventually implment, but the on-chip graphics of these two? Nope.

**Comment by ghost** [comment]
[Citygate D30 XT] Hedaka HED-919 (Undocumented upper memory handling)
[OPTi 283] Olivetti M300-xx (One of them)
[ACC 2168] AMI 386DX clone (Generic AMI board)
[ACC 2168] Packard Bell PB410A (Only for it's integrated Video 7 chip but chipset is undocumented)
[UMC 8881] A-Trend ATC-1415 (Generic Award)
[UMC 8881] ECS Elite UM8810PAIO (Generic Phoenix that hates our FDC665)
[UMC 889x] Shuttle HOT-539 (Undocumented chipset)
[i430VX] Azza 5IVG (Generic Award with just Prime3C utilization)
[i440BX] Gigabyte GA-686BX (Generic Award)
[VIA Apollo ProMedia] Jetway 603TCF (VT8601 has an integrated GPU)

I go against removing the 5598 boards though considering the boards didn't came with support of it's integrated VGA

**Comment by OBattler** [comment]
> [Citygate D30 XT] Hedaka HED-919 (Undocumented upper memory handling)

This one it seems we rretty much have a consensus to remove.

> [OPTi 283] Olivetti M300-xx (One of them)

I already removed both on my end, and explained why in EngiNerd's PR.

> [ACC 2168] AMI 386DX clone (Generic AMI board)
> [ACC 2168] Packard Bell PB410A (Only for it's integrated Video 7 chip but chipset is undocumented)

Sure, but the latter is a machine that has gotten years of testing, plus we have Zorix with the real thing. And as fo the AMI 386DX clone, we don't have many 386DX machines at all.

> [UMC 8881] A-Trend ATC-1415 (Generic Award)
> [UMC 8881] ECS Elite UM8810PAIO (Generic Phoenix that hates our FDC665)
> [UMC 889x] Shuttle HOT-539 (Undocumented chipset)

On these we pretty much have a consensus to remove them, so yes.

> [i430VX] Azza 5IVG (Generic Award with just Prime3C utilization)

I agree this can go, yes.

> [i440BX] Gigabyte GA-686BX (Generic Award)

This was added because of PCem, and I guess it can stay for the same reason as the Shuttle HOT-557.

> [VIA Apollo ProMedia] Jetway 603TCF (VT8601 has an integrated GPU)

Then this can indeed go.

> I go against removing the 5598 boards though considering the boards didn't came with support of it's integrated VGA

It would not be a problem if it was not for the fact it has to be completely redone by me. This is the thing - reverse-engineered chipsets that have been there for years and are known to have a sensible (if partial) implementation are less of a problem than complex reverse-engineered chipsets that have to be redone.

**Comment by ghost** [comment]
More to remove

AMI, Phoenix and Quadtel IBM AT's (It's unconfirmed if the BIOSes were even meant for the IBM AT at all. Quadtel for example has a speed option that even the normal IBM AT can't do as it wasn't intended for it)

[ISA] Micronics 386 Clone (Tandon based board with really unknown jumper handling. Also lot's of undocumented ports and unconfirmed if the BIOS were meant for that board as it normally comes with Phoenix)
[ALi M1429G] Acer A1G (Scraps the M1429 off completely. Replace with Acer V10)
[i430FX] Intel Advanced/ATX (MR) (Only fact it was the only MR BIOS Pentium we had. It's useless now)

[OPTi 495] Award 486 Clone (Controversial removal but do we know anything about this board? Do we know if it's 386/486 combo? We don't even know what that U-Board actually is)

One of the Socket 4 Dells (They use the same BIOS)

[i430FX] QDI Chariot (Not removal but find the proper name for it. Same for the VAIO)

**Comment by OBattler** [comment]
> [ISA] Micronics 386 Clone (Tandon based board with really unknown jumper handling. Also lot's of undocumented ports and unconfirmed if the BIOS were meant for that board as it normally comes with Phoenix)

Than can then go, yes.

> [ALi M1429G] Acer A1G (Scraps the M1429 off completely. Replace with Acer V10)

And where do you propose I find the Acer V10 BIOS?

> [i430FX] Intel Advanced/ATX (MR) (Only fact it was the only MR BIOS Pentium we had. It's useless now)

Maybe that can go, yes.

> [OPTi 495] Award 486 Clone (Controversial removal but do we know anything about this board? Do we know if it's 386/486 combo? We don't even know what that U-Board actually is)

We don't, really. But we then need a non-PCI Award 4.50 486 and an Award 4.50 386DX to replace it.

> One of the Socket 4 Dells (They use the same BIOS)

I guess that can be considered.

> [i430FX] QDI Chariot (Not removal but find the proper name for it. Same for the VAIO)

It's the QDI FMB, and we also know its real Super I/O chip.

**Reply by ghost** [reply]
> We don't, really. But we then need a non-PCI Award 4.50 486 and an Award 4.50 386DX to replace it.

FIC ELI 6. Details can be found on it's UH19 page below

http://www.win3x.org/uh19/motherboard/show/6830

386DX is really complicated considering there are only UMC 491 BIOS for it which are ofc undocumented, while there's also a simplistic SARC one. Both out of question.

**Reply by OBattler** [reply]
The FIC ELI 6 uses the VIA VT82C486 chipset which has no datasheet available. Though, this point is now moot as the existing Award 4.50 machine is going to stay.

**Comment by ghost** [comment]
Being the owner for the UMC 486 implementations, I removed them all on my rework PR and added the PC Chips M919 in exchange. It'll stay as is till more documentation about the chipset or even a 1995 UMC databook comes to surface.
